viewing_party/party.py

```diff
@@ -1,23 +1,175 @@
+import pytest
+
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    
+    if title and genre and rating:
+            return {
+                "title": title,
+                "genre": genre,
+                "rating": rating
+            }
+    return None
+
+
+def add_to_watched(user_data, movie):
+    user_data['watched'].append(movie)  
+    return user_data
+    
+
+def add_to_watchlist(user_data, movie):
+    user_data['watchlist'].append(movie)
+    return user_data
+
+
+def watch_movie(user_data,title):
+    movie_to_move = None
+    for movie in user_data['watchlist']:
+        if movie['title'] == title:
+            movie_to_move = movie
+            break
+          
+    if movie_to_move:
+        user_data['watchlist'].remove(movie_to_move)
+        user_data['watched'].append(movie_to_move)
+    
+    return user_data
+          
 
-# -----------------------------------------
 # ------------- WAVE 2 --------------------
-# -----------------------------------------
 
+def get_watched_avg_rating(user_data):
+
+    watched_movies = user_data.get('watched', [])
+
+    rating_sum = 0.0
+    total_movies = 0
+
+    for movie in watched_movies:
+        rating = movie.get('rating')
+        if rating is not None:
+            rating_sum += rating
+            total_movies += 1
+
+    if total_movies == 0:
+        return 0.0  
+
+    average_rating = rating_sum / total_movies
+    return average_rating
+
+
+def get_most_watched_genre(user_data):
+
+    watched_movie = user_data.get('watched',[])
+    
+    #count occurences of each genre
+    genre_count = {}
+    most_watched_genre = ""
+    max_count = 0
+     
+    for movie in watched_movie:
+        genre = movie.get('genre', None)
+
+        if genre:
+            if genre not in genre_count:
+                genre_count[genre] = 1
+            else:           
+                genre_count[genre] += 1
+
+        if genre_count[genre] > max_count:
+            max_count = genre_count[genre]
+            most_watched_genre = genre
+
+    if watched_movie == []:
+        return None
 
-# -----------------------------------------
+    return most_watched_genre
+        
+               
 # ------------- WAVE 3 --------------------
-# -----------------------------------------
+
+def get_unique_watched(user_data):
+    user_unique_watched_movies = []
+    watched_movie = user_data.get('watched',[])
+    friends = user_data.get('friends',[])

```

Comment:
I think that I would be okay with `user_data["watched"]` syntax here. Mainly because the on requirements of the data structure then all instances of `user_data` should have keys `friends` and `watched`. If not, then the caller of the function should take another look at the data structure.